services/web/server/src/simcore_service_webserver/projects/db.py

```diff
@@ -368,6 +370,10 @@ async def list_projects(  # pylint: disable=too-many-arguments
                     projects.join(projects_to_products, isouter=True)
                     .join(access_rights_subquery, isouter=True)
                     .join(folders_to_projects, isouter=True)
+                    if settings.WEBSERVER_FOLDERS
+                    else projects.join(projects_to_products, isouter=True).join(
+                        access_rights_subquery, isouter=True
+                    )
```

Comment:
you are moving logic into the db layer here.
It would make more sense to make folder_id optional when the folders is disabled.

Reply:
The issue is folder ID is already option, it means a root directory.